Project/Source/Components/ComponentBilboardRender.cpp

```diff
@@ -0,0 +1,209 @@
+#include "ComponentBilboardRender.h"
+
+#include "GameObject.h"
+#include "Components/ComponentTransform.h"
+#include "Components/UI/ComponentButton.h"
+#include "Application.h"
+#include "Modules/ModulePrograms.h"
+#include "Modules/ModuleCamera.h"
+#include "Modules/ModuleRender.h"
+#include "Modules/ModuleEditor.h"
+#include "Modules/ModuleResources.h"
+#include "Modules/ModuleTime.h"
+#include "Modules/ModuleUserInterface.h"
+#include "Panels/PanelScene.h"
+#include "Resources/ResourceTexture.h"
+#include "Resources/ResourceShader.h"
+#include "FileSystem/TextureImporter.h"
+#include "FileSystem/JsonValue.h"
+#include "Math/float3x3.h"
+#include "Utils/ImGuiUtils.h"
+#include "Math/TransformOps.h"
+#include "imgui.h"
+#include "GL/glew.h"
+#include "debugdraw.h"
+
+#include "Utils/Leaks.h"
+
+#define JSON_TAG_TEXTURE_SHADERID "ShaderId"
+#define JSON_TAG_TEXTURE_TEXTUREID "TextureId"
+#define JSON_TAG_INITCOLOR "InitColor"
+#define JSON_TAG_FINALCOLOR "FinalColor"
+
+#define JSON_TAG_COLOR "Color"
+#define JSON_TAG_YTILES "Ytiles"
+#define JSON_TAG_XTILES "Xtiles"
+#define JSON_TAG_ANIMATIONSPEED "AnimationSpeed"
+
+void ComponentBilboardRender::Update() {
+	Draw();
+}
+
+void ComponentBilboardRender::Init() {
+}
+
+void ComponentBilboardRender::DrawGizmos() {
+}
+
+void ComponentBilboardRender::OnEditorUpdate() {
+	if (ImGui::Checkbox("Active", &active)) {
+		if (GetOwner().IsActive()) {
+			if (active) {
+				Enable();
+			} else {
+				Disable();
+			}
+		}
+	}
+	ImGui::Separator();
+
+	ImGui::TextColored(App->editor->textColor, "Texture Settings:");
+	ImGui::Separator();
+
+	ImGui::Checkbox("Random Frame", &isRandomFrame);
+	ImGui::ResourceSlot<ResourceShader>("shader", &shaderID);
+
+	UID oldID = textureID;
+	ImGui::ResourceSlot<ResourceTexture>("texture", &textureID);
+
+	ResourceTexture* textureResource = App->resources->GetResource<ResourceTexture>(textureID);
+
+	if (textureResource != nullptr) {
+		int width;
+		int height;
+		glGetTextureLevelParameteriv(textureResource->glTexture, 0, GL_TEXTURE_WIDTH, &width);
+		glGetTextureLevelParameteriv(textureResource->glTexture, 0, GL_TEXTURE_HEIGHT, &height);
+
+		if (oldID != textureID) {
+			ComponentTransform2D* transform2D = GetOwner().GetComponent<ComponentTransform2D>();
+			if (transform2D != nullptr) {
+				transform2D->SetSize(float2(static_cast<float>(width), static_cast<float>(height)));
+			}
+		}
+
+		ImGui::Text("");
```

Comment:
FYI: Exists ImGui::NewLine() if this is the purpose

Reply:
Also Dummy() lets you control the exact spacing that you need